monix-reactive/shared/src/main/scala/monix/reactive/internal/operators/SwitchMapObservable.scala

```diff
@@ -61,15 +63,24 @@ private[reactive] final class SwitchMapObservable[A, B](source: Observable[A], f
           activeChild := childObservable.unsafeSubscribeFn(new Observer[B] {
             def onNext(elem: B) =
               self.synchronized {
-                if (upstreamIsDone || myChildIndex != activeChildIndex)
+                if (myChildIndex != activeChildIndex)
                   Stop
                 else {
                   ack = out.onNext(elem).syncOnStopOrFailure(_ => cancelFromDownstream())
```

Comment:
BTW it seems like this `ack` var is unused other than this two lines

Reply:
It's set below to `Stop` in `cancelFromDownstream`